src/Workspaces/Core/Portable/Workspace/Solution/ProjectInfo.cs

```diff
@@ -380,7 +385,12 @@ internal class ProjectAttributes : IChecksummedObject, IObjectWritable
                 Id = id ?? throw new ArgumentNullException(nameof(id));
                 Name = name ?? throw new ArgumentNullException(nameof(name));
                 Language = language ?? throw new ArgumentNullException(nameof(language));
-                AssemblyName = assemblyName ?? throw new ArgumentNullException(nameof(assemblyName));
+
+                // Empty assembly names are not allowed.  For now just default to an unused assembly
+                // name.
+                AssemblyName = string.IsNullOrWhiteSpace(assemblyName)
+                    ? InvalidAssemblyNamePrefix + Interlocked.Increment(ref s_nextInvalidAssemblySuffix)
+                    : assemblyName;
```

Comment:
Well, that's the issue we're running into today.  We're throwing an exception, and it's causing problems :D